Dalamud/ImGuiScene/Helpers/ImGuiViewportHelpers.cs

```diff
@@ -0,0 +1,164 @@
+    public unsafe delegate Vector2* GetWindowSizeDelegate(Vector2* returnStorage, ImGuiViewportPtr viewport);
+    
+    /// <summary>
+    /// Delegate to be called when the window should be given focus.
+    /// </summary>
+    /// <param name="viewport">An instance of <see cref="ImGuiViewportPtr"/>.</param>
+    public delegate void SetWindowFocusDelegate(ImGuiViewportPtr viewport);
+
+    /// <summary>
+    /// Delegate to be called when the window is focused.
+    /// </summary>
+    /// <param name="viewport">An instance of <see cref="ImGuiViewportPtr"/>.</param>
+    /// <returns>Whether the window is focused.</returns>
+    public delegate bool GetWindowFocusDelegate(ImGuiViewportPtr viewport);
+
+    /// <summary>
+    /// Delegate to be called when whether the window is minimized is queried.
+    /// </summary>
+    /// <param name="viewport">An instance of <see cref="ImGuiViewportPtr"/>.</param>
+    /// <returns>Whether the window is minimized.</returns>
+    public delegate bool GetWindowMinimizedDelegate(ImGuiViewportPtr viewport);
+
+    /// <summary>
+    /// Delegate to be called when the window title should be changed.
+    /// </summary>
+    /// <param name="viewport">An instance of <see cref="ImGuiViewportPtr"/>.</param>
+    /// <param name="title">The new title.</param>
+    public delegate void SetWindowTitleDelegate(ImGuiViewportPtr viewport, string title);
+
+    /// <summary>
+    /// Delegate to be called when the window alpha should be changed.
+    /// </summary>
+    /// <param name="viewport">An instance of <see cref="ImGuiViewportPtr"/>.</param>
+    /// <param name="alpha">The new alpha.</param>
+    public delegate void SetWindowAlphaDelegate(ImGuiViewportPtr viewport, float alpha);
+
+    /// <summary>
+    /// Delegate to be called when the IME input position should be changed.
+    /// </summary>
+    /// <param name="viewport">An instance of <see cref="ImGuiViewportPtr"/>.</param>
+    /// <param name="pos">The new position.</param>
+    public delegate void SetImeInputPosDelegate(ImGuiViewportPtr viewport, Vector2 pos);
+
+    /// <summary>
+    /// Delegate to be called when the window's DPI scale value is queried.
+    /// </summary>
+    /// <param name="viewport">An instance of <see cref="ImGuiViewportPtr"/>.</param>
+    /// <returns>The DPI scale.</returns>
+    public delegate float GetWindowDpiScaleDelegate(ImGuiViewportPtr viewport);
+
+    /// <summary>
+    /// Delegate to be called when viewport is changed.
+    /// </summary>
+    /// <param name="viewport">An instance of <see cref="ImGuiViewportPtr"/>.</param>
+    public delegate void ChangedViewportDelegate(ImGuiViewportPtr viewport);
+
+    /// <summary>
+    /// Disables ImGui from disabling alpha for Viewport window backgrounds.
+    /// </summary>
+    public static unsafe void EnableViewportWindowBackgroundAlpha()
+    {
+        var offset = 0x00007FFB6ADA632C - 0x00007FFB6AD60000;
```

Comment:
This is a little scary, we should ideally fix it in our imgui fork. Maybe leave a todo so that we can do that when we rebuild it some day?

Reply:
Ideally we would remove this line:
https://github.com/goatcorp/gc-imgui/blob/e4920dc10cbf32c0cd9b4a48a4a71c8814242460/imgui.cpp#L6126